java/org/apache/catalina/webresources/AbstractSingleArchiveResourceSet.java

```diff
@@ -131,6 +141,31 @@ protected boolean isMultiRelease() {
         return multiRelease.booleanValue();
     }
 
+    @Override
+    public void gc() {
```

Comment:
When I traced this out, this call gets made a lot (like a few dozen times) on what feels like every request.  I think that's bad because it does two things:

1. synchronizes on the acthiveLock...yet again
2. clears out the  `archiveEntries` which forces the classloader to reopen and reread that jar on next access (synchronized)

I don't think closing the archive is important here, since I'm closing the jar in a `finally` in here after reading the contents.

Reply:
It looks like the implementation of `closeJarFile` is just:
```
    protected void closeJarFile() {
        synchronized (archiveLock) {
            archiveUseCount--;
        }
    }
```
That is, it seems like it's using reference counting and it's not actually closed except by `gc()`?

Reply:
Good catch...I wasn't sure what I wanted to do with this yet, mostly just learning how it does business.